opennms-web-api/src/main/java/org/opennms/web/api/Util.java

```diff
@@ -164,9 +164,10 @@ protected static String substituteUrl(final HttpServletRequest request, final St
      */
     public static String getHostHeader(final HttpServletRequest request) {
         for (int i = 0; i < hostHeaders.length; ++i) {
-        	final String ret = request.getHeader(hostHeaders[i]);
-            if (ret != null) {
-                return ret;
+            // Get the first value in the header (support for proxy-chaining)
+            final Enumeration<String> headers = request.getHeaders(hostHeaders[i]);
+            if (headers.hasMoreElements()) {
+                return headers.nextElement();
```

Comment:
The behavior appears to be identical: we're still using the value from the first occurrence of the header.

I understand that the value of the X-Forwarded-Host header may many (comma separated) values, but I don't see how this change solves the problem.
Aren't the values in X-Forwarded-Host 
It still looks like we're using the value from the first occurence